### ISSUE TITLE
New test failure

### ISSUE THREAD

**Issue by opoplawski** [Unidata/netcdf4-python#662]
Starting to get build failures - https://apps.fedoraproject.org/koschei/package/netcdf4-python?collection=f27

Failure is:
```
+ PYTHONPATH=../build/lib.linux-x86_64-2.7
+ /usr/bin/python2 run_all.py
...............................................F............E.............................
======================================================================
ERROR: runTest (tst_enum.EnumTestCase)
testing enum data type
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builddir/build/BUILD/netcdf4-python-1.2.7rel/test/tst_enum.py", line 36, in setUp
    cloud_var[:] = datain_masked
  File "netCDF4/_netCDF4.pyx", line 4059, in netCDF4._netCDF4.Variable.__setitem__ (netCDF4/_netCDF4.c:46083)
ValueError: trying to assign illegal value to Enum variable
======================================================================
FAIL: runTest (tst_masked.PrimitiveTypesTestCase)
testing auto-conversion of masked arrays and packed integers
----------------------------------------------------------------------
not running tst_unicode3.py ...
netcdf4-python version: 1.2.7
HDF5 lib version:       1.8.18
netcdf lib version:     4.4.1.1
numpy version           1.13.0rc1
Traceback (most recent call last):
  File "/builddir/build/BUILD/netcdf4-python-1.2.7rel/test/tst_masked.py", line 111, in runTest
    assert(file['v'][0] is NP.ma.masked)
AssertionError
----------------------------------------------------------------------
Ran 90 tests in 31.707s
FAILED (failures=1, errors=1)
```
Not much change from the previous successful build so not sure what is up.  What other information would be useful?

**Comment by jswhit** [comment]
I'm guessing its numpy related.  Do the tests pass with numpy 1.12.1?

**Comment by jswhit** [comment]
I confirmed that these failures are related to changes in the way masked arrays are handled in numpy 1.13.  Pull request coming soon.

**Comment by jswhit** [comment]
Looks to me like this is due to a regression in numpy 1.13.rc0.  I've reported this in https://github.com/numpy/numpy/issues/9121.

**Comment by jswhit** [comment]
Fix for failing tst_enum.py in pull request #663 - tst_masked.py still fails because of change in behaviour of scalar numpy masked arrays in 1.13.

**Comment by jswhit** [comment]
OK, received clarification from the numpy folks that this change was intentional and implemented their suggested fix in pull request #663.  Tests pass now with numpy 1.13.0rc1.

**Comment by jswhit** [comment]
merging pull request now...
